File: 백준/Silver/2606. 바이러스/바이러스.py
import sys
from collections import deque
n = int(sys.stdin.readline())
v = int(sys.stdin.readline())

graph = [[] for i in range(n+1)] # 그래프 초기화
visited=[0]*(n+1) # 방문한 컴퓨터인지 표시
for i in range(v): # 그래프 생성
    a,b=map(int,input().split())
    graph[a]+=[b] # a에 b 연결
    graph[b]+=[a] # b에 a 연결 -> 양방향
visited[1]=1 # 1번 컴퓨터부터 시작이니 방문 표시
Q=deque([1])
while Q:
    c=Q.popleft()
    for nx in graph[c]:
        if visited[nx]==0:
            Q.append(nx)
            visited[nx]=1

sys.stdout.write(f"{sum(visited)-1}")

sys.stdout.flush() 

File: 백준/Silver/2606. 바이러스/바이러스.rs
use std::io ::{BufReader,BufWriter,BufRead,Write,stdin,stdout};
use std::collections::VecDeque;


fn main(){
    let mut reader= BufReader::new(stdin().lock());
    let mut writer= BufWriter::new(stdout().lock());
    let mut input = String::new();

    reader.read_line(&mut input).unwrap();
    //컴퓨터 개수
    let n:usize= input.trim().parse().unwrap();
    input.clear();
    reader.read_line(&mut input).unwrap();
    //연결선 개수
    let v:usize= input.trim().parse().unwrap();
    
     //그래프  초기화
    let mut graph: Vec<Vec<usize>> = vec![vec![]; n + 1];
    //방문한 컴퓨터인지 확인
    let mut visited: Vec<usize> = vec![0; n + 1];
    
    for _ in 0..v {
        input.clear();
        reader.read_line(&mut input).unwrap();
        let nums: Vec<usize> = input
            .split_whitespace()
            .map(|x| x.trim().parse().unwrap())
            .collect();
        let a = nums[0];//a에 b연결
        let b = nums[1];//b에 a연결 -> 양방향
        graph[a].push(b);
        graph[b].push(a);
    }
     //1번 컴퓨터
    visited[1] = 1;
    let mut queue = VecDeque::new();
    queue.push_back(1);

    while let Some(c) = queue.pop_front() {
        for &nx in &graph[c] {
            if visited[nx] == 0 {
                queue.push_back(nx);
                visited[nx] = 1;
            }
        }
    }

    let total_visited = visited.iter().sum::<usize>() - 1;
    writeln!(writer,"{}", total_visited).unwrap();
}